third_party/hashtree/hashtree_source.BUILD

```diff
@@ -0,0 +1,139 @@
+# BUILD file for building hashtree from source with proper cross-compilation support
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+package(default_visibility = ["//visibility:public"])
+
+# Build hashtree library for AMD64 - only build when targeting x86_64
+genrule(
```

Comment:
At first blush, it looks really fishy to me to be using genrules to cross-compile C code.
Doesn't prysm get compiled for a target architecture?
If so, then, we should only need to compile the library for that architecture as well. Right?
If so, you should be able to use the `cc_library` rule and not worry about cross-compiling.

What am I missing?